### PR TITLE
WIP: Add Support for Probability Calibration Curve

### LINKED ISSUE #365: Probability Calibration Curve
When performing classification one often wants to predict not only the class label but also the associated probability to give a level of confidence in the prediction. The `sklearn.metrics.calibration_curve` method returns true and predicted probabilities that can be visualized across multiple models to make the best model selection based on reliability. 

Probability Calibration Curve

- [Scikit-Learn Example: Probability Calibration Curves](http://scikit-learn.org/stable/auto_examples/calibration/plot_calibration_curve.html#probability-calibration-curves)

Alexandru Niculescu-Mizil and Rich Caruana (2005) Predicting Good Probabilities With Supervised Learning, in Proceedings of the 22nd International Conference on Machine Learning (ICML). See section 4 (Qualitative Analysis of Predictions).

### PR DESCRIPTION
**Pull request by decreaseme**
Fixes #365 

https://github.com/DistrictDataLabs/yellowbrick/issues/365

Changes proposed in this pull request:

- Create a new feature for generating calibration curves via class `CalibrationCurve` (subclassing ClassificationScoreVisualizer) 
-  Implements the `fit()`, `score()`, and `poof()` method that creates a 1. main calibration curve graph and 2. mean predicted value line graph.

will follow similar pattern as http://www.scikit-yb.org/en/latest/api/classifier/class_balance.html

@DistrictDataLabs/team-oz-maintainers

### CODE REVIEW

**Comment by lwgray** [comment]
Thank you for joining us for the Yellowbrick Sprint at PyCon Cleveland. And also, thank you for submitting your first PullRequest - oh yeah! 

We will review this asap. We have been quite busy with all the great contributions from sprinters like yourself.  In the meantime, feel free to work on another issue.  We hope you continue to contribute.

**Comment by bbengfort** [comment]
@decreaseme can't wait to see what you come up with for this!

**Comment by decreaseme** [comment]
yup! i'll hopefully get it done by this weekend :)

**Comment by mattharrison** [comment]
Excited for this. Came by to file a feature request, but saw that it was already being worked on. Awesome!

**Comment by bbengfort** [comment]
@decreaseme let me know if I can answer any questions!

**Comment by decreaseme** [comment]
Will do! Sorry it's been taking longer than expected.

**Comment by lwgray** [comment]
@decreaseme If you have any questions or are stuck with a particular piece of code.... just shoot us a message.  We are more than happy to help.

**Comment by rebeccabilbro** [comment]
Hey there @decreaseme - thanks so much for reaching out about this one! Since we've had a new version bump of YB since you opened it and things have been a little quiet here, I'm going to go ahead and close this, though I'd encourage you to open a new one as soon as you're ready to jump back in! Let us know!
